vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java

```diff
@@ -560,6 +626,144 @@ public void mouseReleased(MouseEvent e) {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        moveDownButton.doClick();
+      }
+    });
+
+    inUseList.getInputMap(JComponent.WHEN_FOCUSED).put(
+      KeyStroke.getKeyStroke(KeyEvent.VK_HOME, SwingUtils.getModifierKeyMask()), "TopInUse"); //$NON-NLS-1$
+    inUseList.getActionMap().put("TopInUse", new AbstractAction() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        moveTopButton.doClick();
+      }
+    });
+
+    inUseList.getInputMap(JComponent.WHEN_FOCUSED).put(
+      KeyStroke.getKeyStroke(KeyEvent.VK_END, SwingUtils.getModifierKeyMask()), "BottomInUse"); //$NON-NLS-1$
+    inUseList.getActionMap().put("BottomInUse", new AbstractAction() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        moveBottomButton.doClick();
+      }
+    });
+
+    inUseList.getInputMap(JComponent.WHEN_FOCUSED).put(
+      KeyStroke.getKeyStroke(KeyEvent.VK_LEFT, SwingUtils.getModifierKeyMask()), "ToAvail"); //$NON-NLS-1$
+    inUseList.getInputMap(JComponent.WHEN_FOCUSED).put(
+      KeyStroke.getKeyStroke(KeyEvent.VK_LEFT, 0), "ToAvail"); //$NON-NLS-1$
+    inUseList.getInputMap(JComponent.WHEN_FOCUSED).put(
+      KeyStroke.getKeyStroke(KeyEvent.VK_RIGHT, SwingUtils.getModifierKeyMask()), "ToAvail"); //$NON-NLS-1$
+    inUseList.getInputMap(JComponent.WHEN_FOCUSED).put(
+      KeyStroke.getKeyStroke(KeyEvent.VK_RIGHT, 0), "ToAvail"); //$NON-NLS-1$
+    inUseList.getActionMap().put("ToAvail", new AbstractAction() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        availableList.requestFocus();
+      }
+    });
+
+    // Use typed keys to jump to entries in trait list
+    inUseList.addKeyListener(new KeyAdapter() {
+      private long time;
+
+      @Override
+      public void keyTyped(KeyEvent e) {
+        final char ch = e.getKeyChar();
+
+        //ignore searches for non alpha-numeric characters
+        if (!Character.isLetterOrDigit(ch)) {
+          return;
+        }
+
+        final String key = Character.toString(Character.toLowerCase(ch));
+
+        // Iterate through items in the list until a matching prefix is found.
+        for (int i = 0; i < inUseList.getModel().getSize(); i++) {
+
+          final int index = (i + inUseList.getSelectedIndex() + 1) % inUseList.getModel().getSize();
+
+          final GamePiece trait = inUseList.getModel().getElementAt(index);
+          final String str;
+
+          if (trait instanceof EditablePiece) {
+            str = ((EditablePiece)trait).getDescription();
+          }
+          else {
+            final String s = trait.getClass().getName();
+            str = s.substring(s.lastIndexOf('.') + 1);
+          }
+
+          if (str.toLowerCase().startsWith(key)) {
+            inUseList.setSelectedIndex(index);     //change selected item in list
+            inUseList.ensureIndexIsVisible(index); //change listbox scroll-position
+            break;
+          }
+        }
+      }
```

Comment:
This looks duplicated from above. Could both be refactored into a function?

Reply:
I'm not quite sure how? (I mean yes I'm sure it's possible, I just am not sure how to do it)